File: Web/src/main/java/com/itechart/webflux/web/config/jwt/JwtAuthToken.java
package com.itechart.webflux.web.config.jwt;

import com.itechart.webflux.web.core.model.entity.UserRole;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import java.util.List;

public class JwtAuthToken extends UsernamePasswordAuthenticationToken {

    private String token;
    private String username;
    private String role;

    public JwtAuthToken(String token) {
        super(null, null);
        this.token = token;
    }

    public JwtAuthToken(String token, Object principal, Object credentials, List<GrantedAuthority> grantedAuthorities) {
        super(principal, credentials, grantedAuthorities);
        role = credentials instanceof UserRole ? ((UserRole) credentials).name() : credentials.toString();
        username = (String) principal;
        this.token = token;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @Override
    public Object getCredentials() {
        return role;
    }

    @Override
    public Object getPrincipal() {
        return username;
    }

}

File: Web/src/main/java/com/itechart/webflux/web/config/jwt/JwtAuthToken.kt
package com.itechart.webflux.web.config.jwt

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.authority.SimpleGrantedAuthority
import java.util.*


open class JwtAuthToken(var token: String?, private var username: String?, private var role: String?) : UsernamePasswordAuthenticationToken(username, role, Collections.singleton(SimpleGrantedAuthority(role))) {

    constructor(token: String?) : this(token, null, null)

    override fun getCredentials(): Any? {
        return role
    }

    override fun getPrincipal(): Any? {
        return username
    }


}